ifrs17/Report/ReportMutableScopesInteractive.ipynb

```diff
@@ -0,0 +1,1094 @@
+                "\n                        .ExecuteAsync();",
+                "\n    }",
+                "\n    ",
+                "\n",
+                "\n    async Task<GridOptions> GetReportTaskAsync(IDataCube<ReportVariable> data) {",
+                "\n        return await report.ForDataCube(data)",
+                "\n            .WithQuerySource(workspace)",
+                "\n            .SliceRowsBy(rowSlices)",
+                "\n            .SliceColumnsBy(columnSlices)",
+                "\n            .ReportGridOptions(headerColumnWidth: headerColumnWidthValue)",
+                "\n            .ExecuteAsync();",
+                "\n    }",
+                "\n",
+                "\n    void Init(){",
+                "\n        var task = InitReportStorageScopeAsync();",
+                "\n        task.Wait();",
+                "\n    }",
+                "\n",
+                "\n    async Task InitReportStorageScopeAsync() { // This has the Async issue, but imo it should come in the future",
+                "\n       await GetStorage().InitializeReportIndependentCacheAsync();",
+                "\n    }",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "# Report Scopes"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Best Estimate PV"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "[InitializeScope(nameof(Init))]",
+                "\npublic interface PvReport : ReportScope {",
+                "\n    ",
+                "\n    IDataCube<ReportVariable> ReportScope.GetData() => GetScopes<LockedBestEstimate>(GetDataIdentities()).Aggregate().LockedBestEstimate + GetScopes<CurrentBestEstimate>(GetDataIdentities()).Aggregate().CurrentBestEstimate;",
+                "\n",
+                "\n     void Init() {",
+                "\n         // BasicSliceAndDiceFormsEntity",
+                "\n         defaultRowSlices = new string[] { nameof(ReportVariable.Novelty), nameof(ReportVariable.VariableType) };",
+                "\n         defaultColumnSlices = new string[] { nameof(ReportVariable.Currency), nameof(ReportVariable.EconomicBasis) };",
+                "\n    }",
+                "\n",
+                "\n    private List<string> specificFilters => new List<string> {nameof(ReportVariable.AmountType)};",
+                "\n    IReadOnlyCollection<string> GetFilterName() => defaultFilters.ToList().Concat(specificFilters).ToArray();",
```

Comment:
Can we avoid this ToList()? Maybe change the original type to List already?

Reply:
This is already done, so I am closing this comment.